internal/pkg/cli/app_deploy.go

```diff
@@ -0,0 +1,423 @@
+	GetEnvironments(projectName string) ([]*archer.Environment, error)
+}
+
+func (i *appDeployInput) init() error {
+	projectService, err := project.NewService()
+	if err != nil {
+		return fmt.Errorf("create project service: %w", err)
+	}
+	i.projectService = projectService
+
+	toolsAccountSession, err := session.Default()
+	if err != nil {
+		return fmt.Errorf("initialize tools account session: %w", err)
+	}
+	i.ecrService = ecr.New(toolsAccountSession)
+
+	workspaceService, err := workspace.New()
+	if err != nil {
+		return fmt.Errorf("intialize workspace service: %w", err)
+	}
+	i.workspaceService = workspaceService
+
+	return nil
+}
+
+func (i *appDeployInput) sourceInputs() error {
+	if err := i.sourceProjectName(); err != nil {
+		return err
+	}
+
+	if err := i.sourceProjectData(); err != nil {
+		return err
+	}
+
+	if err := i.sourceAppName(); err != nil {
+		return err
+	}
+
+	if err := i.sourceEnvName(); err != nil {
+		return err
+	}
+
+	if err := i.sourceImageTag(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i *appDeployInput) sourceProjectName() error {
+	projects, err := i.projectService.GetProjects()
+
+	if err != nil {
+		return err
+	}
+
+	if len(projects) == 0 {
+		// TODO: recommend project init? or archer init?
+		return errors.New("no projects found, try ./archer init")
+	}
+
+	projectNames := []string{}
+
+	for _, p := range projects {
+		projectNames = append(projectNames, p.Name)
+	}
+
+	if i.project == "" {
+		if len(projects) == 1 {
+			projectName := projects[0].Name
+
+			// NOTE: defaulting project value, tell the user.
+			fmt.Printf("Only found one project, defaulting to: %s\n", projectName)
+
+			i.project = projectName
+
+			return nil
+		}
+
+		selectedProjName, err := i.prompt.SelectOne("Select a project", "", projectNames)
+
+		if err != nil {
+			return fmt.Errorf("select project: %w", err)
+		}
+
+		i.project = selectedProjName
+	}
+
+	for _, p := range projects {
+		if p.Name == i.project {
+			return nil
+		}
+	}
+
+	return errors.New("invalid project name")
+}
+
+func (i *appDeployInput) sourceProjectData() error {
+	apps, err := i.projectService.GetApplications(i.project)
+
+	if err != nil {
+		return fmt.Errorf("get apps: %w", err)
+	}
+
+	if len(apps) == 0 {
+		// TODO: recommend follow up command - app init?
+		return errors.New("no applications found")
+	}
+
+	i.projectApplications = apps
+
+	envs, err := i.projectService.GetEnvironments(i.project)
+
+	if err != nil {
+		return fmt.Errorf("get projects: %w", err)
+	}
+
+	if len(envs) == 0 {
+		// TODO: recommend follow up command - env init?
+		fmt.Printf("couldn't find any environments associated with project %s, try initializing one: `archer env init`\n", i.project)
+
+		return errors.New("no environments found")
+	}
+
+	i.projectEnvironments = envs
+
+	return nil
+}
+
+func (i *appDeployInput) sourceAppName() error {
+	appNames := []string{}
+
+	for _, app := range i.projectApplications {
```

Comment:
Should we only show local apps (using `ws.AppNames`)? Just because we can't really deploy to an application that isn't in the workspace.

Reply:
I'm not sure I follow. An application would have to be associated with the project in SSM to be deployable?

Reply:
Ah I see. 

The only things that are eligible for deployment are the applications that are within a workspace AND have been registered in SSM. 

My comment was that we should restrict the list of apps to the ones which are in the workspace (we can also validate that they are in the SSM). But if there are 3 apps and 2 of them are in my workspace, there's no need to display the 3rd since it's source isn't local (we won't be able to build it).